### DISCUSSION TITLE
How to control specular...

### DISCUSSION THREAD

**Discussion by Honya2000** [google/filament#4847] [General]
Hello,

I'm exporting GLB from Blender. Blender Principled BSDF shader doesn't show any reflections if specular and metalsess are 0.
But in filament specular is very strong if both parameters are 0, because of DFG lookup which depends only on roughness and view angle. 

In my material roughness is 0.4. But both metalness and reflectivity are 0.
In this case lookup from DFG gives value 0.2 if F0 == 0.0 and NoV == 0.0;
   vec3 dfg = textureLod(light_iblDFG, vec2(NoV, lod), 0.0).rgb;
   return mix(dfg.xxx, dfg.yyy, pixel.f0);

For roughness 1.0 it gives almost 1.0 regardless of F0 value because both dfg.x and dfg.y == 1 on DFG UV's == 0.0 .

Question is why reflectivity and metalness almost do not affect IBL specular? Why is material so shiny when both parameters are 0?
So I cannot make filament PBR materials compatible with blender PBR GLB export ?

**Comment by romainguy** [comment]
Setting specular is not supported by gltf without extensions (Filament does not support that extension ). Metalness by itself does not affect reflectivity. Non-metallic objects have a specular component.

**Comment by romainguy** [comment]
https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_specular

Note that I would argue setting specular to 0 is not physically plausible.

**Comment by Honya2000** [comment]
Problem is not in GLTF at all. I modified GLTF exporter in blender. It exports every material attribute now.
Loader also was modified.

I'm talking about reflectance part of the material (which is specular factor of blender Principled BSDF):
    float reflectance = computeDielectricF0(material.reflectance);
    pixel.f0 = computeF0(baseColor, material.metallic, reflectance);

Here I even hardcoded f0 to zero. It still shiny like christmass tree. I cannot control specular at all, cannot remove it:
This is the example where reflectivity and metalness both zero:

![image](https://user-images.githubusercontent.com/34769255/141695564-d4c0a523-025c-4c47-93c4-bfa3144d7c44.png)

**Comment by Honya2000** [comment]
For now I'm using specular color to control reflectness in indirect lighting. But this is hacky way:

    vec3 E = specularDFG(pixel) * material.specularColor.rgb;

**Comment by romainguy** [comment]
Check out the [materials documentation](https://google.github.io/filament/Materials.html). Setting `reflectance` to 0 will not remove specular highlights (and again, `metallic` has nothing to do with this). And `reflectance` is not strictly equal to the `specular` parameter in Blender. Filament does not let you remove the specular component of the BRDF. Like I mentioned earlier, you need support for `KHR_materials_specular` to do what you want, and Filament doesn't implement it yet.

You can use the specularGlossiness material model instead to kill the specular component by setting the `specularColor` to 0. It's however a different workflow than metallic/roughness.

**Comment by Honya2000** [comment]
The thing is I need PBR shader model, but with specular control.
For now will use my hack. It is compatible with blender now.
